src/server.js

```diff
@@ -107,13 +107,10 @@ function setupRenderer() {
         console.error('ROUTER ERROR:', pretty.render(error));
         res.status(500);
         hydrateOnClient();
-      } else if (!renderProps) {
-        res.status(500);
-        hydrateOnClient();
-      } else {
+      } else if (renderProps) {
         const component = (
           <Provider store={store} key="provider">
-            <RoutingContext />
+            <RoutingContext {...renderProps} />
```

Comment:
@dts Adding the `renderProps` into the RoutingContext here can cause endless looping issues (see #16) with the current version of redux-simple-router. We can't safely add that in until React Router 2.0 changes are resolved. Otherwise, I like the 404 follow up.